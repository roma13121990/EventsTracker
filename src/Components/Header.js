import React from 'react'

export default class Header extends React.Component {
    constructor(props){
        super(props);
        this.state = {value: ''};
    }

    handleChange(event) {
        if (event.key === 'Enter') {
            this.setState({value: event.target.value},() => {
                this.props.onSearch(this.state.value);
            });
        }
        else{
            this.setState({value: event.target.value});
        }
    }

    render() {
        return (
            <header className="App-header row">
                <a className="App-logo col-xs-6 col-md-6 col-sm-6 col-lg-6" href="https://www.bandsintown.com/?came_from=257">
                    <img src="https://assets.bandsintown.com/images/bitFist.svg" alt="Bandsintown Fist"/>
                    <img src="https://assets.bandsintown.com/images/bitText.svg" alt="Bandsintown Text"
                         className="Logo-text"/>
                </a>
                <div className="Searchbar col-xs-6 col-md-6 col-sm-6 col-lg-6">
                    <div>
                        <img src="https://assets.bandsintown.com/images/loupe.svg" alt="" />
                        <input type="text" className="value" placeholder="eg. Maroon 5" onKeyPress={(e) => this.handleChange(e)}/>
                    </div>
                </div>
            </header>)
    }
}