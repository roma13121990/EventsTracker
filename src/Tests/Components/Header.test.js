import React from 'react'
import renderer from 'react-test-renderer'
import Header from "../../Components/Header";
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({adapter: new Adapter()});

describe('<Header />', () => {

    it('Should generate snapshot for Header component', () => {
        const tree = renderer.create(<Header />).toJSON();
        expect(tree).toMatchSnapshot()
    })

    it('Should call handleChange on keyPress on input', () => {
        const wrapper = mount(<Header />);
        wrapper.find(".value").simulate("keyPress", {
            target: {
                value: "Maroon 5"
            }
        });
        expect(wrapper.instance().state).toEqual({"value": "Maroon 5"})
    })

    it('Should call handleChange on keyPress on input when enter is pressed', () => {
        const onSearch = jest.fn();
        const wrapper = mount(<Header onSearch={onSearch} />);

        wrapper.find(".value").simulate("keyPress", {key:"Enter",
            target: {
                value: "Maroon 5"
            }
        });
        expect(wrapper.instance().state).toEqual({"value": "Maroon 5"})
        expect(onSearch).toHaveBeenCalled();
    })
});