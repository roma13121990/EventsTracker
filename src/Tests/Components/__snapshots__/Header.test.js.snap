// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should generate snapshot for Header component 1`] = `
<header
  className="App-header row"
>
  <a
    className="App-logo col-xs-6 col-md-6 col-sm-6 col-lg-6"
    href="https://www.bandsintown.com/?came_from=257"
  >
    <img
      alt="Bandsintown Fist"
      src="https://assets.bandsintown.com/images/bitFist.svg"
    />
    <img
      alt="Bandsintown Text"
      className="Logo-text"
      src="https://assets.bandsintown.com/images/bitText.svg"
    />
  </a>
  <div
    className="Searchbar col-xs-6 col-md-6 col-sm-6 col-lg-6"
  >
    <div>
      <img
        alt=""
        src="https://assets.bandsintown.com/images/loupe.svg"
      />
      <input
        className="value"
        onKeyPress={[Function]}
        placeholder="eg. Maroon 5"
        type="text"
      />
    </div>
  </div>
</header>
`;
